common/src/main/kotlin/org/imdc/extensions/common/DatasetExtensions.kt

```diff
@@ -220,8 +224,40 @@ object DatasetExtensions {
     @Suppress("unused")
     @ScriptFunction(docBundlePrefix = "DatasetExtensions")
     @KeywordArgs(
-        names = ["input", "headerRow", "sheetNumber", "firstRow", "lastRow", "firstColumn", "lastColumn"],
-        types = [ByteArray::class, Integer::class, Integer::class, Integer::class, Integer::class, Integer::class, Integer::class],
+        names = ["dataset", "filterNull"],
+        types = [Dataset::class, Boolean::class],
+    )
+    fun toDict(args: Array<PyObject>, keywords: Array<String>): PyStringMap {
+        val parsedArgs = PyArgParser.parseArgs(
+            args,
+            keywords,
+            arrayOf("dataset", "filterNull"),
+            Array(2) { Any::class.java },
+            "toDict",
+        )
+        val dataset = parsedArgs.requirePyObject("dataset").toJava<Dataset>()
+        val filterNull = parsedArgs.getBoolean("filterNull").orElse(false)
+        return PyStringMap(
+            dataset.columnIndices.associate { col ->
+                dataset.getColumnName(col) to PyList(
+                    buildList {
+                        for (row in dataset.rowIndices) {
+                            val value = dataset[row, col]
+                            if (value != null || !filterNull) {
+                                add(value)
+                            }
+                        }
+                    },
+                )
```

Comment:
Technically this allocates one unnecessary Pair instance for each iteration, but since the actual heavy object here will be the list of column data, I think this is fine. `getColumnAsList` would be helpful here, but then you'd have to filter the list as a second operation, meaning two scans of the same column instead of just one.